src/main/scala/org/broadinstitute/dsde/agora/server/business/AgoraBusiness.scala

```diff
@@ -61,34 +63,54 @@ class AgoraBusiness(permissionsDataSource: PermissionsDataSource)(implicit ec: E
     }
   }
 
+  //Workaround for https://github.com/broadinstitute/wdl4s/issues/103
+  //Checks to see if the WDL has imports, since loading a WDL that contains imports without specifying a resolver
+  //barfs in an unintuitive way.
+  private def checkWdlHasNoImports(wdl: String): Try[Unit] = {
+    Try { //AST loader might syntax error if the wdl is crap
+      val ast = AstTools.getAst(wdl, "string")
+      ast.getAttribute("imports").asInstanceOf[AstList].isEmpty
+    } match {
+      case Failure(x) => Failure(x) //ast didn't parse nicely
+      case Success(true) => Success() //no imports, WDL is fine
+      case Success(false) => Failure(ValidationException("Agora doesn't support imports in WDLs"))
+    }
+  }
+
   private def checkValidPayload[T](agoraEntity: AgoraEntity, username: String)(op: => ReadWriteAction[T]): ReadWriteAction[T] = {
-    Try(agoraEntity.entityType.get match {
-
-      case AgoraEntityType.Task =>
-      // GAWB-59 remove wdl validation
-      //        val namespace = WdlNamespace.load(agoraEntity.payload.get, BackendType.LOCAL)
-      //        // Passed basic validation.  Now check if (any) docker images that are referenced exist
-      //        namespace.tasks.foreach { validateDockerImage }
-
-      case AgoraEntityType.Workflow =>
-      // GAWB-59 remove wdl validation
-      //        val resolver = MethodImportResolver(username, this)
-      //        val namespace = WdlNamespace.load(agoraEntity.payload.get, resolver.importResolver _, BackendType.LOCAL)
-      //        // Passed basic validation.  Now check if (any) docker images that are referenced exist
-      //        namespace.tasks.foreach { validateDockerImage }
-
-      case AgoraEntityType.Configuration =>
-        val json = agoraEntity.payload.get.parseJson
-        val fields = json.asJsObject.getFields("methodRepoMethod")
-        if(fields.size != 1) throw new ValidationException("Configuration payload must define at least one field named 'methodRepoMethod'.")
-
-        val subFields = fields(0).asJsObject.getFields("methodNamespace", "methodName", "methodVersion")
-        if(!subFields(0).isInstanceOf[JsString]) throw new ValidationException("Configuration methodRepoMethod must include a 'methodNamespace' key with a string value")
-        if(!subFields(1).isInstanceOf[JsString]) throw new ValidationException("Configuration methodRepoMethod must include a 'methodName' key with a string value")
-        if(!subFields(2).isInstanceOf[JsNumber]) throw new ValidationException("Configuration methodRepoMethod must include a 'methodVersion' key with a JSNumber value")
-    }) match {
+    val wdl = agoraEntity.payload.get
+    val wdlImportCheck = checkWdlHasNoImports(wdl)
```

Comment:
why not inline wdlImportCheck? As it stands now, it also runs, although ignored, for a configuration.